### ISSUE TITLE
ER: Inconsistent filtering behavior for "Languages/Technologies/Tools" category

### ISSUE THREAD

**Issue by jphamtv** [hackforla/website#6709]
### Emergent Requirement - Problem

1. The filter behavior for the 'Languages/Technologies/Tools' category on the Home and Projects pages is confusing and inconsistent. For example, when selecting 1 language and 1 technology, the filter uses OR logic, but when selecting 2 items from the same category, the filter uses AND logic.

2. There are duplicate Languages and Tools categories on the Home page.

    Homepage filter:

    <img width="400" alt="Screenshot 2024-04-23 at 12 21 51" src="https://github.com/hackforla/website/assets/4952258/36d811e3-c035-4179-9aaf-7f3499177ceb">

### Issue you discovered this emergent requirement in
- #6467 
- #6549 

### Date discovered 
4/22/24

### Did you have to do something temporarily
- [ ] YES
- [x] NO

### Who was involved
@jphamtv

### What happens if this is not addressed
Users may select filters that don't match their intended search, leading to incorrect results.

### Resources
[www.hackforla.org](https://www.hackforla.org/#)
[www.hackforla.org/projects](https://www.hackforla.org/projects/)
[current-projects.js](https://github.com/hackforla/website/blob/gh-pages/assets/js/current-projects.js)

### Recommended Action Items
- [x] Make a new issue
- [x] Discuss with team
- [ ] Let a Team Lead know

### Potential solutions [draft]
After discussing with the Dev Lead team, the solution is to create the following 2 issues:

- Issue 1 - Medium - Remove the extra Languages and Tools categories from the homepage to match the Projects page.
- Issue 2 - Large - Update filters to consistently use AND logic across all filter categories and within the same filter category. Ensure that the search function's OR logic continues to work as expected when multiple keywords are entered in the 'Search by Keywords' field (for example, 'Python OR JavaScript' returns all projects that use Python or JavaScript).

**Comment by ExperimentsInHonesty** [comment]
@jphamtv A couple of things have come up since we talked yesterday.

1. There is already an ER for the search tips and its both comprehensive and less comprehensive at the same time. https://github.com/hackforla/website/issues/6327
2. After reviewing the other issue, I think the solution could be better than having the text visible.  So here is what I suggest.  I suggest that you post a comment on the other issue with the screenshots and the following changes and then remove issue 3 from this ER. 

## Proposed solution
Add link below the Filter section that opens a modal that has the following text and table

#### Link Text
Search Tips

#### Modal Content
To find projects matching any of your interests, use Boolean Operators in the Search Bar.  See the table below to understand more how Boolean Operators work.  

Boolean Operators | Meaning | Example
-- | -- | --
**AND** | Limit results | React AND Node (Search for project cards that contain both React and Node.)
**OR** | One term OR another | Python OR Javascript (Search for project cards that contains Python OR JavaScript.)
**-** | Exclude a term from the search | React -Django (Limits project card results to only those with React and not the term Django.)


#### Link Text placement on Desktop/Tablet:
<img width="400" alt="Screenshot 2024-04-23 at 12 20 23" src="https://github.com/hackforla/website/assets/4952258/e17fc692-b814-4754-bc3c-0d0ed0675ed6">
    
#### Link Text placement on Mobile:    
<img width="379" alt="Screenshot 2024-04-23 at 13 14 43" src="https://github.com/hackforla/website/assets/4952258/d2f63b83-3c97-43c6-8f77-0554c6d0ca33">

**Comment by jphamtv** [comment]
@ExperimentsInHonesty - Removed Issue 3 from this ER, and posted a comment on #6327 with the screenshots and changes as suggested in the previous comment.

**Comment by marioantonini** [comment]
1. Availability: M-W 2-5pm
2. ETA: EoD 05/21/2024

**Comment by marioantonini** [comment]
Update:
1. Availability: W-Th, Sat-Sun 2-5pm
2. ETA: EoD 05/26/2024

**Comment by marioantonini** [comment]
1. Availability: Sat-Sun 2-5pm
2. ETA: EoD 06/09/2024

**Comment by marioantonini** [comment]
1. Availability: Sun 2-6 pm, Tue, Wed 1-4pm
2. ETA: EoD 06/25/2024

**Comment by k-cardon** [comment]
Hi @ExperimentsInHonesty, Roslyn suggested I might be able to work on this issue (or alternatively #6145) in order to make a medium issue for myself to work on (and a large issue for the backlog). Is that acceptable?

**Comment by HackforLABot** [comment]
Hi @k-cardon, thank you for taking up this issue! Hfla appreciates you :)

Do let fellow developers know about your:-
i. **Availability:** (When are you available to work on the issue/answer questions other programmers might have about your issue?)
ii. **ETA:** (When do you expect this issue to be completed?)

You're awesome!

P.S. - You may not take up another issue until this issue gets merged (or closed). Thanks again :)

**Comment by k-cardon** [comment]
Availability: usually Thursday evening and weekends
ETA: not sure if I'm permitted to work on this yet-- @roslynwythe  asked me to check with @ExperimentsInHonesty before starting.

**Comment by k-cardon** [comment]
I'm going to remove myself from this since I'm not sure if I'm permitted to work on it, and there are medium issues available now.

**Comment by HackforLABot** [comment]
Hello @pluto-bell, we appreciate you taking on this issue, however it looks like you're already working on another issue at this time. Please wait until your current issue is merged before taking on another issue. :)

We are going to unassign you from this issue so you can focus on your current issue.

Hfla appreciates you! :)

**Comment by HackforLABot** [comment]
Hello @gmgonzal, we appreciate you taking on this issue, however it looks like you're already working on another issue at this time. Please wait until your current issue is merged before taking on another issue. :)

We are going to unassign you from this issue so you can focus on your current issue.

Hfla appreciates you! :)

**Comment by HackforLABot** [comment]
Hi @caz002, thank you for taking up this issue! Hfla appreciates you :)

Do let fellow developers know about your:-
i. **Availability:** (When are you available to work on the issue/answer questions other programmers might have about your issue?)
ii. **ETA:** (When do you expect this issue to be completed?)

You're awesome!

P.S. - You may not take up another issue until this issue gets merged (or closed). Thanks again :)

**Comment by caz002** [comment]
Availability: After 7PM on Weekdays
ETA: 10/1/2025

**Comment by caz002** [comment]
@t-will-gillis Is it okay to work on Issue 1 for this ER to meet the Medium requirement, given that Issue 2 is Large?

**Comment by t-will-gillis** [comment]
@caz002 If you would like to write both of these issues, and then after the issues are approved you can assign yourself to work on one of them. Let us know what you think

**Comment by caz002** [comment]
@t-will-gillis I am interested in writing both issues, thank you!

**Comment by caz002** [comment]
### Created Medium Issue:
- https://github.com/hackforla/website/issues/8352

**Comment by caz002** [comment]
### Created Large Issue:
- https://github.com/hackforla/website/issues/8353

**Comment by daras-cu** [comment]
@caz002 the issues you created look good, I am going to close this issue and assign you to the medium one you created. Thanks for working on this!

**Comment by caz002** [comment]
Great to hear, thank you for reviewing!
